api/src/distribution/route.rs

```diff
@@ -301,6 +304,57 @@ pub async fn create_distribution_release(
         StatusCode::INTERNAL_SERVER_ERROR
     })?;
 
+    // Extract services from packages after transaction commits (non-fatal)
+    let packages = sqlx::query_as!(
+        Package,
+        "SELECT * FROM package WHERE id = ANY($1)",
+        &distribution_release.packages as &[i32]
+    )
+    .fetch_all(&state.pg_pool)
+    .await;
+
+    if let Ok(packages) = packages {
+        for pkg in packages {
+            if pkg.file.ends_with(".deb") {
+                match Storage::download_from_s3(&state.config.packages_bucket_name, &pkg.file).await
+                {
+                    Ok(data) => match extract_services_from_deb(&data) {
+                        Ok(services) => {
+                            for service in services {
+                                if let Err(e) = sqlx::query!(
+                                    "INSERT INTO release_services (release_id, package_id, service_name, watchdog_sec)
```

Comment:
I guess this assumes that everything that has a service should be checked. Perhaps it would be nice to have some services not be checked. Not sure though